tests/testthat/test-expand.R

```diff
@@ -46,11 +46,26 @@ test_that("preserves ordered factors", {
   expect_equal(df$a, ordered("a"))
 })
 
-test_that("zero length inputs are automatically dropped", {
+
+test_that("zero length numeric & character inputs are automatically dropped", {
   tb <- tibble::tibble(x = 1:5)
   expect_equal(expand(tb, x, y = numeric()), tb)
   expect_equal(nesting(x = tb$x, y = numeric()), tb)
   expect_equal(crossing(x = tb$x, y = numeric()), tb)
+
+  expect_equal(expand(tb, x, y = character()), tb)
+  expect_equal(nesting(x = tb$x, y = character()), tb)
+  expect_equal(crossing(x = tb$x, y = character()), tb)
+})
+
+test_that('zero length factor inputs are completed by expand & crossing, dropped by nesting', {
```

Comment:
Why does nesting drop?

Reply:
This is a fantastic question and one that I should have noted.  Per the docs:

> nesting() is the complement to crossing(): it only keeps combinations of all variables that appear in the data

As a result, I was not sure if an empty factor should be considered a "combination appearing in the data."  If so, I am happy to work on an implementation.  I can see where this might come across as a lack of consistency - however, this is a context where `nesting` / `crossing` are designed to behave differently. 

For instance, non-empty factors are currently handled differently by `nesting` and `crossing.` My aim was to maintain that dichotomy:

```r
tb <- tibble::tibble(x=1:2)                       
emptyfactor <- factor(levels = c("a","b"))        
                                                  
simplefactor <- factor(c("a"),levels = c("a","b"))
                                                  
## nesting factor behavior                        
nesting(x = tb$x, y = simplefactor)               
#> # A tibble: 2 x 2
#>       x      y
#>   <int> <fctr>
#> 1     1      a
#> 2     2      a
                                                  
## crossing factor behavior                       
crossing(x = tb$x, y = simplefactor)              
#> # A tibble: 4 x 2
#>       x      y
#>   <int> <fctr>
#> 1     1      a
#> 2     1      b
#> 3     2      a
#> 4     2      b
```

Reply:
Yeah, I don't see how you can "nest" the empty levels of a factor because they don't appear in the data. I think this is the issue at the heart of #193